jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -147,8 +147,7 @@ public ProcessDefinition buildProcessDefinition(String deploymentId,String bpmn2
             BPMN2ProcessFactory.setBPMN2ProcessProvider(provider);
         }
         try {
-            ProcessDescRepoHelper threadLocalHelper = new ProcessDescRepoHelper();
-	        BPMN2DataServiceSemanticModule.setRepoHelper(threadLocalHelper);
+            ProcessDescRepoHelper threadLocalHelper = BPMN2DataServiceSemanticModule.getRepoHelper();
```

Comment:
The main change is actually the one onder here (Overriding the `ThreadLocal<ProcessDescRepoHelper>.get()` method in `BPMN2DataServiceSemanticModule`). 

If you change the code back to the way it was and run the attached test, you'll see an NPE caused by the race condition. The race condition here is that the ` BPMN2ProcessFactory.setBPMN2ProcessProvider(provider)` method sets the provider for *all* threads, while the repo helper is only set for the local threads. 

Because the data service `BPMN2ProcessProvider` implementation references handlers that use the repo helper (which, in the case of other threads, is null), an NPE is thrown. I've seen this NPE regularly during startup of kie-wb.